src/Symfony/Component/PropertyAccess/Exception/UnexpectedTypeException.php

```diff
@@ -18,8 +18,4 @@
  */
 class UnexpectedTypeException extends RuntimeException
 {
-    public function __construct($value, $expectedType)
```

Comment:
Yep, it is a theoretical BC break, hence my note at the bottom of the original PR comment. I don't see a straightforward way around it:

- Using a new subclass won't work because the constructor is already overloaded.
- Using another exception class entirely would be a worse BC break.
- Adding optional constructor arguments might be a smaller BC break, but those arguments would make no sense when the exception is used in `PropertyPath`.

I can of course change this somehow, or add the BC break notes, or scrap the PR entirely.